File: pkg/pwrgrep/testdata/fixtures/c/side-effects.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

void f(void) {
    // ruleid: c-side-effects
    system("ls");
    // ruleid: c-side-effects
    FILE *out = fopen("out.txt", "w");
    // ok: c-side-effects
    FILE *in = fopen("in.txt", "r");
    // ruleid: c-side-effects
    unlink("tmp");
    // ruleid: c-side-effects
    int s = socket(AF_INET, SOCK_STREAM, 0);
}

File: pkg/pwrgrep/testdata/fixtures/csharp/SideEffects.cs
public class SideEffects
{
    async Task F(HttpClient client, SqlConnection conn)
    {
        // ruleid: cs-side-effects
        Process.Start("ls");
        // ruleid: cs-side-effects
        File.Delete("tmp");
        // ruleid: cs-side-effects
        await client.GetAsync("https://example.com");
        // ruleid: cs-side-effects
        var cmd = new SqlCommand("SELECT 1", conn);
        // ruleid: cs-side-effects
        cmd.ExecuteReader();
        // ok: cs-side-effects
        Console.WriteLine("hi");
    }
}
